### PR TITLE
feat(ai-training-jobs): add possibility to attach ssh keys

### PR DESCRIPTION
**Pull request by mael-le-gal**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MLS-2250
| License          | BSD 3-Clause

## Description

On AI Training add the possibility to attach an ssh key when submiting a new job

### :flags: Translations

- [x] TRANS-41398

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/training/training.constants.js
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/training/training.constants.js:6]: i think we can avoid the '_CONSTANTS' word suffix, we are already into a const file.

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js
Comment (mael-le-gal) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js]: fixed

File: packages/manager/modules/pci/src/projects/project/training/training.constants.js
Comment (mael-le-gal) [reply packages/manager/modules/pci/src/projects/project/training/training.constants.js]: fixed

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js:357]:
```suggestion
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:481]:
```suggestion
                <div class="oui-field__control py-2">
```

**Review by antleblanc — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js
Comment (mael-le-gal) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js]: fixed

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html
Comment (mael-le-gal) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html]: fixed

**Review by antleblanc — DISMISSED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by marie-j — DISMISSED** [review]

**Review by marie-j — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Comment by mael-le-gal** [comment]
QA done on this one. Needs approvals before merging.

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js:106]:
```suggestion
    // Option indicating if the user wants to add ssh keys
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js:355]:
```suggestion
    return this.PciProjectTrainingService.getSavedSshKeys(this.projectId).then(
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:424]:
```suggestion
                                data-translate="pci_projects_project_training_jobs_submit_select_ssh_legend"
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:434]: **issue(blocking):** Due to the ngRepeat (L420), this will ended up with some duplicate IDs.

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:463]: **issue(blocking):** Looks like this translation is coming from a copy/paste. We should used a scoped translation.

**Review by antleblanc — CHANGES_REQUESTED** [review]
Thanks for the update @mael-le-gal!

I've left some comments/requests changes.
Could you please have a look?

Thanks!

**Comment by mael-le-gal** [comment]
Thank's for the comments @antleblanc. I made some corrections.

**Review by antleblanc — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_fr_FR.json
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_fr_FR.json:80]: To keep consistency
```suggestion
  "pci_projects_project_training_jobs_submit_select_ssh_delete_message": "Supprimer la clé SSH attachée"
```

**Review by JeremyDec — APPROVED** [review]

**Review by marie-j — DISMISSED** [review]

**Review by antleblanc — APPROVED** [review]

**Comment by antleblanc** [comment]
Pulling translations.

**Comment by antleblanc** [comment]
Rebasing on `develop` in order to solve existing conflicts:
```txt
packages/manager/modules/pci/src/projects/project/training/jobs/info/translations/Messages_de_DE.json
packages/manager/modules/pci/src/projects/project/training/jobs/info/translations/Messages_en_GB.json
packages/manager/modules/pci/src/projects/project/training/jobs/info/translations/Messages_es_ES.json
packages/manager/modules/pci/src/projects/project/training/jobs/info/translations/Messages_it_IT.json
packages/manager/modules/pci/src/projects/project/training/jobs/info/translations/Messages_pl_PL.json
packages/manager/modules/pci/src/projects/project/training/jobs/info/translations/Messages_pt_PT.json
```

**Comment by antleblanc** [comment]
Rebased successfully on `develop` branch.

**Pull request merged**
